test/unit/xml_description_test.py

```diff
@@ -1,67 +1,109 @@
-
-from mock import patch
-
 import mock
+from builtins import bytes
 
 from .test_helper import *
 
 from kiwi.exceptions import (
     KiwiSchemaImportError,
     KiwiValidationError,
     KiwiDescriptionInvalid,
-    KiwiDataStructureError
+    KiwiDataStructureError,
+    KiwiDescriptionConflict
 )
 from kiwi.xml_description import XMLDescription
 
 
 class TestSchema(object):
     def setup(self):
-        self.description = XMLDescription('description')
+        test_xml = bytes(
+            b"""<?xml version="1.0" encoding="utf-8"?>
+            <image schemaversion="1.4" name="bob">
+                <description type="system">
+                    <author>John Doe</author>
+                    <contact>john@example.com</contact>
+                    <specification>
+                        say hello
+                    </specification>
+                </description>
+                <preferences>
+                    <packagemanager>zypper</packagemanager>
+                    <version>1.1.1</version>
+                    <type image="ext3"/>
+                </preferences>
+                <repository type="rpm-md">
+                    <source path="repo"/>
+                </repository>
+            </image>"""
+        )
+        self.description_from_file = XMLDescription(description='description')
+        self.description_from_data = XMLDescription(xml_content=test_xml)
+
+    @raises(KiwiDescriptionConflict)
+    def test_constructor_conflict(self):
+        XMLDescription(description='description', xml_content='content')
+
+    def test_load_schema_from_xml_content(self):
+        parsed = self.description_from_data.load()
+        assert parsed.get_schemaversion() == '6.3'
```

Comment:
Maybe the schema version can be picked up from somewhere, so we don't need to fix this test all the time XSL template changes? (Just a thought)